### PR TITLE
Add table of testing configurations

### PR DESCRIPTION
**Pull request by agocke**
This table should make it clear what we expect we are testing on, and can be used to compare with our YAML and confirm that it's doing the same thing.

### CODE REVIEW

**Comment by agocke** [comment]
Note: I filled in a few items, but need some help

**Comment by agocke** [comment]
cc @jkotas

**Review by hoyosjs — APPROVED** [review]

**Comment by lambdageek** [comment]
I'm not sure this way of organizing the document is particularly illuminating.

In particular "libraries tests" and "mono tests" don't make a lot of sense to me: I assume the former means the tests under `src/libraries/**/tests` whereas the latter mean... `src/tests` running using Mono?

I would suggest grouping by RID and then by Runtime /Libraries / Installer tests and then listing the execution engine configurations that run those tests.

## Linux x64

- Runtime tests
  - CoreCLR
  - Mono JIT, interp, llvmaot, llvmfullaot
- Libraries tests
  - CoreCLR
  - Mono JIT, interp, llvmaot, llvmfullaot
- Installer

## windows-x64
- Runtime tests
  - CoreCLR
- Libraries tests
  - CoreCLR 

## iossimulator-arm64
- Runtime tests
  - Mono interp
- Libraries tests
  - Mono interp

## browser-wasm
- Runtime tests
  - Mono interp, llvmfullaot
- Libraries tests
  - Mono interp

etc

**Comment by radical** [comment]
Just for reference, this is where we have documented the builds for wasm - https://github.com/dotnet/runtime/tree/main/src/mono/wasm#how-do-i-know-which-jobs-run-on-ci-and-when

**Comment by steveisok** [comment]
> https://github.com/dotnet/runtime/tree/main/src/mono/wasm#how-do-i-know-which-jobs-run-on-ci-and-when

Wasm points out configurations that run smoke tests on PR's.  Would be a good idea to spell that out for the rest.

**Comment by agocke** [comment]
@lambdageek No, that isn't meant to be directory based, it's meant to differentiate between tests that are meant to test the runtime, as opposed to tests for libraries. So Mono tests would be testing the mono runtime, and libraries tests would be testing library correctness. There's some crossover here, but strict hierarchy/ontology isn't useful. You can put the data in whatever section feels most appropriate.

File: docs/infra/test-configurations.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+### Libraries Tests
+
+### Installer Tests
+
+All builds are
+
+- Windows 10 x64 Client VS2019
+  - x64 Release
+  - Cross x86 Release
```
[the remark this answers is not in the document]
Comment (agocke) [reply docs/infra/test-configurations.md:31]: It's mainly pointing out that the architecture being built/tested is different than the machine, but you're right that that might not be useful

**Comment by agocke** [comment]
@radical @steveisok I don't think this is quite what I'm looking for. I'm trying to define the platform coverage for our CI, so that we know what platforms we're testing on, and roughly what versions. So in this case, it's fine to say "Linux" if what you're building/testing is basically "linux-unknown", but I would expect to at least know what the current representative for "linux-unknown" is, i.e. in the docs above it's Ubuntu 18.04 x64

**Comment by agocke** [comment]
Merged this for now -- we can fill in more as we go.

**Comment by agocke** [comment]
I haven't gone through all the libraries tests, but I expect we'll see 3 or 4 different linux distributions being tested -- and that's great, because then we can look at the coverage and see what we've run tests on, and what's unknown

**Pull request merged**